sdk/cosmos/src/headers/mod.rs

```diff
@@ -58,3 +58,13 @@ pub(crate) fn add_partition_keys_header(
     let serialized = partition_keys.to_json();
     builder.header(HEADER_DOCUMENTDB_PARTITIONKEY, serialized)
 }
+
+pub(crate) fn add_header<T: azure_core::AddAsHeader>(
```

Comment:
At some point we should move this to core and just make it public. All service clients will need this, though most common headers should just be configuration options when building a pipeline. Worth moving to core now?

Reply:
We don't expose headers directly to users. Is there a reason users need to concern themselves with headers instead of setting more abstract options on request builders?

Reply:
Users may not need to (though, they should be able to via policy), but other clients should. In general, most of the standard headers are handled by the pipeline builders. Clients (the SDKs we'll be writing) will be adding or modifying headers as needed (e.g. If-Modified-Since). So won't this need to be `pub` instead of `pub(crate)`? azure_core and azure_<everything else> will be different crates.

Reply:
The underlying trait is already public (along with the implementing structs) so it makes sense to make this pub as well. Or maybe we can just move it inside the trait?